extensions/cornerstone/src/OHIFCornerstoneViewport.js

```diff
@@ -221,6 +239,7 @@ class OHIFCornerstoneViewport extends Component {
           viewportIndex={viewportIndex}
           imageIds={imageIds}
           imageIdIndex={currentImageIdIndex}
+          onNewImage={this.onNewImage}
```

Comment:
Using this hook to update state used to degrade performance, but it looks like your recent enhancements may have resolved that issue.